### DISCUSSION TITLE
is that usefull or portable for hantek6074bd

### DISCUSSION THREAD

**Discussion by gumby0q** [Ho-Ro/Hantek6022API#24] [Q&A] [answered]
Hi,
I bought hantek6074bd device like 5 years ago but rarely use it due to ugly software. I have been looking for alternatives and there are not so many. More over I would like to try make some interface that will be convenient for touch screen.

So far to rich my goal I see the only option to use hantek dll files and make some app based on them.
 
I didn't dive much in specifics of Hantek6022 and Hantek6022API but can anybody give me advice:
1. Are Hantek6022 and hantek6074bd similar in approach which requires uploading firmware to scope after the scope was connected? If so how to investigate that?
2. Can this repository be useful for porting some stuff to at least kickstart hantek6074bd working with linux/android?
3. Any thoughts about that and how complex that is?

**Comment by Ho-Ro** [comment, marked as answer]
Sorry to disappoint you, but despite the identical housing both scopes work differently. The 6022 has a Cypress EzUSB chip that does the complete work in firmware - setup USB connection and stream 16 bit values to USB with a max usable speed of ~24 MByte/s. The 16 bit values come directly from two 8 bit ports that are connected to a dual 8 bit ADC (plus some analog signal conditioning). The faster Hanteks, e.g. your 6074 have a FPGA in between the EzUSB and the analog frontend that does the high speed sampling and buffering. The buffered data are transferred with lower speed to USB. So you would have to write or reverse engineer your Hantek FW and adapt the OH6022 SW, because it is highly optimised for use with this direct USB streaming described above.
A possible (but also not so easy) solution would be to start with [openhantek](https://github.com/OpenHantek/openhantek), the predecessor project of OpenHAntek6022. It was unmaintained for years, but there was some activity during the last weeks.
